### PR TITLE
PMM-4466 migrate database to postgres

### PR DESCRIPTION
**Pull request by nikita-b**
Custom branches: 
pmm-update: https://github.com/percona/pmm-update/pull/157
pmm-server: https://github.com/percona/pmm-server/pull/314
jenkins-pipeline: https://github.com/Percona-Lab/jenkins-pipelines/pull/1272

### CODE REVIEW

File: ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    db: grafana
+    name: grafana
+    password: grafana
+    priv: 'ALL'
+    expires: infinity
+    state: present
+  when: not ansible_check_mode
+
+- name: Create actual grafana config
+  include_role:
```
Comment (atymchuk) [inline ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml:26]: How will this work once we migrate this role to Go (pmm-update)?

File: ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    name: 'grafana'
+    state: stopped
+
+- name: Install grafana-migrate
+  yum:
+    name: "grafana-db-migrator"
+    state: latest
+
+- name: Run grafana migrator
+  command: grafana-migrate /srv/grafana/grafana.db "postgres://grafana:grafana@localhost:5432/grafana?sslmode=disable"
```
Comment (atymchuk) [inline ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml:49]: Maybe better call it `grafana-migrate-db` to make it obvious?

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml (hunk 2 above)
Comment (nikita-b) [reply ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml]: Renamed

File: ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml (hunk 1 above)
Comment (nikita-b) [reply ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml:26]: We run this role from Ansible from pmm-update. It's the same Ansible.

File: ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  yum:
+    name: "grafana-db-migrator"
+    state: latest
+
+- name: Run grafana migrator
+  command: grafana-db-migrator /srv/grafana/grafana.db "postgres://grafana:grafana@localhost:5432/grafana?sslmode=disable"
+  register: migrator_output
+  changed_when: "'All done' in migrator_output.stdout"
+
+- name: start grafana again
```
Comment (nikita-b) [inline ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml:53]:
```suggestion
- name: Start grafana again
```

File: ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  supervisorctl:
+    name: grafana
+    state: started
+
+- name: create backup for SQLite Grafana database
```
Comment (nikita-b) [inline ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml:58]:
```suggestion
- name: Create backup for SQLite Grafana database
```

File: ansible/playbook/tasks/update.yml [hunk 5]
```diff
@@ -34,6 +34,31 @@
...
+      when: ansible_virtualization_type != "docker"
+      service:
+        name: supervisord
+        state: started
+        enabled: yes
+
+    - name: Check that supervisor socket exists
+      stat:
+        path: /var/run/supervisor/supervisor.sock
+      register: is_supervisor_run
```
Comment (atymchuk) [inline ansible/playbook/tasks/update.yml:52]:
```suggestion
      register: is_supervisor_running
```

File: ansible/playbook/tasks/update.yml [hunk 6]
```diff
@@ -34,6 +34,31 @@
...
+
+    - name: Supervisord start         | Start supervisord for docker
+      when: ansible_virtualization_type == "docker" and not is_supervisor_run.stat.exists
```
Comment (atymchuk) [inline ansible/playbook/tasks/update.yml:55]:
```suggestion
      when: ansible_virtualization_type == "docker" and not is_supervisor_running.stat.exists
```

**Review by atymchuk — CHANGES_REQUESTED** [review]
Sorry, two minor things )

File: ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+- name: Create actual grafana config
+  include_role:
+    name: grafana
+
+- name: Start grafana again
+  supervisorctl:
+    name: grafana
+    state: started
+
+- name: Check if initail data were created
```
Comment (atymchuk) [inline ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml:34]:
```suggestion
- name: Check if initial data were created
```

File: ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  postgresql_query:
+    db: grafana
+    query: SELECT 1 FROM org WHERE id=1
+  retries: 3
+  delay: 10
```
Comment (atymchuk) [inline ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml:39]:
```suggestion
  delay: 3
```

**Review by atymchuk — CHANGES_REQUESTED** [review]

**Review by atymchuk — APPROVED** [review]

File: ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+- name: Create actual grafana config
+  include_role:
+    name: grafana
+
+- name: Start grafana again
+  supervisorctl:
+    name: grafana
+    state: started
+
+- name: Wait for grafana database initialization
```
[the remark this answers is not in the document]
Comment (nikita-b) [reply ansible/playbook/tasks/roles/sqlite-to-postgres/tasks/main.yml]: Fixed

**Pull request merged**